checker/src/org/checkerframework/checker/signedness/SignednessVisitor.java

```diff
@@ -23,6 +22,124 @@ public SignednessVisitor(BaseTypeChecker checker) {
         super(checker);
     }
 
+    /** @return true iff node is a mask operation (&amp; or |) */
+    private boolean isMask(Tree node) {
+        Kind kind = node.getKind();
+
+        return kind == Kind.AND || kind == Kind.OR;
+    }
+
+    /** @return true iff expr is a literal */
+    private boolean isLiteral(ExpressionTree expr) {
+        return expr instanceof LiteralTree;
+    }
+
+    /**
+     * @param obj either an Integer or a Long
+     * @return the long value of obj
+     */
+    private long getLong(Object obj) {
+        if (obj instanceof Integer) {
+            Integer intObj = (Integer) obj;
+            return intObj.longValue();
+        } else {
+            return (long) obj;
+        }
+    }
+
+    /**
+     * Given a masking operation of the form {@code x & maskLit} or {@code x | maskLit}, return true
+     * iff the masking operation results in the same output regardless of the value of the numBits
+     * most significant bits of x. This is if the numBits most significant bits of mask are 0 for
+     * AND, and 1 for OR. For example, assuming that numBits is 4, the following is true about AND
+     * and OR masks:
+     *
+     * <p>{@code x & 0x0... == 0x0...;} {@code x | 0xF... == 0xF...;}
+     *
+     * @param maskKind the kind of mask (AND or OR)
+     * @param numBitsLit the LiteralTree whose value is numBits
+     * @param maskLit the LiteralTree whose value is mask
+     * @return true iff the numBits most significant bits of mask are 0 for AND, and 1 for OR
+     */
+    private boolean maskIgnoresMSB(Kind maskKind, LiteralTree numBitsLit, LiteralTree maskLit) {
+        long numBits = getLong(numBitsLit.getValue());
+        long mask = getLong(maskLit.getValue());
+
+        // If maskLit was an int, then shift mask so that the numBits most significant bits are in the right position
+        if (maskLit.getKind() != Kind.LONG_LITERAL) {
+            mask <<= 32;
+        }
+
```

Comment:
I would remove this line, but add a blank line between lines 74 and 75.  The block from 68 to 74 has the goal of putting the numBits most significant bits as the least significant bits, zeroing out the rest.  Say that from the beginning.  I think it would be better to use one operation for each, one on each arm of the if on line 69 -- I think that would be simpler than the current flow.